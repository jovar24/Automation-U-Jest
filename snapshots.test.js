
var itemStock = [
    { 'id': '1', 'itemName': 'oranges', 'stock': '10' },
    { 'id': '2', 'itemName': 'monkey', 'stock': '8' },
    { 'id': '3', 'itemName': 'flour', 'stock': '13' },
    { 'id': '4', 'itemName': 'oranges', 'stock': '6' },
    { 'id': '5', 'itemName': 'apples', 'stock': '30' },
]

function filterItemStock(arr, key, term) {
    return arr.filter(function (obj) {
        return obj[key] === term
    })
}

test('returns all items with matching id', () => {
    expect(filterItemStock(itemStock, 'id', '1')).toEqual([
        { 'id': '1', 'itemName': 'oranges', 'stock': '10', }
    ])
})
test('returns all items with matching itemName', () => {
    expect(filterItemStock(itemStock, 'itemName', 'oranges')).toEqual([
        { 'id': '1', 'itemName': 'oranges', 'stock': '10' },
        { 'id': '4', 'itemName': 'oranges', 'stock': '6' },

    ])
})

test('returns all items with matching id', () => {
    expect(filterItemStock(itemStock, 'id', '1')).toMatchSnapshot()
})
test('returns all items with matching itemName', () => {
    expect(filterItemStock(itemStock, 'itemName', 'oranges')).toMatchSnapshot()
})