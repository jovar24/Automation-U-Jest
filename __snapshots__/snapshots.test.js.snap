// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all items with matching id 1`] = `
Array [
  Object {
    "id": "1",
    "itemName": "oranges",
    "stock": "10",
  },
]
`;

exports[`returns all items with matching itemName 1`] = `
Array [
  Object {
    "id": "1",
    "itemName": "oranges",
    "stock": "10",
  },
  Object {
    "id": "4",
    "itemName": "oranges",
    "stock": "6",
  },
]
`;
